python/ray/dag/compiled_dag_node.py

```diff
@@ -1694,7 +1711,9 @@ def _is_same_actor(idx1: int, idx2: int) -> bool:
         return not topological_order_exists
 
     def _monitor_failures(self):
-        outer = self
+        import weakref
```

Comment:
Do we need to import `weakref` here in case it has already been imported at the top level?